### DISCUSSION TITLE
Relation between PrivateAssets=all and IncludeAssets / ExcludeAssets

### DISCUSSION THREAD

**Discussion by lonix1** [NuGet/Home#11998] [General]
I created a discussion rather than issue as I'm unsure whether this is feasible. I originally asked this [on SO](https://stackoverflow.com/q/73180516/9971404).

The nuget installation pages for development-only packages (e.g. analysers, build tools), typically show this:
````xml
<PackageReference Include="Foo" Version="1.2.3">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime;build;native;contentfiles;analyzers;...</IncludeAssets>
</PackageReference>
````

For such packages, the [`PrivateAssets`](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets) property is always set to `all`. But the `IncludeAssets` property is always specified too. I was wondering whether it's redundant. I got a nice [answer](https://stackoverflow.com/a/73192877/9971404) from @zivkan who explained it's necessary.

But he raised an interesting point, which is that it's neater to use `<ExcludeAssets>compile</ExcludeAssets>`, or even better: `<PackageReference Include="Foo" Version="1.2.3" ExcludeAssets="compile" />`.

I'd like to do that. I'm wondering whether there are possible problems with that?

And if it's "safe", how about using that format on the nuget pages, and simplifying everyone's project files?

**Comment by murshex** [comment]
I noticed this too, I agree with this

**Comment by lonix1** [comment]
I converted this to [an issue](https://github.com/NuGet/Home/issues/13345), please follow there.

**Comment by nkolev92** [comment]
Copying my response from https://github.com/NuGet/Home/issues/13345#issuecomment-2030801750. 

Using ExcludeAssets technically could lead to a changed behavior if a new asset class was introduced, which now would get included because you're using exclude compile.

Can't think of anything else beyond that.
Related issue: https://github.com/NuGet/Home/issues/8098.
